### PR TITLE
Enable to sign any EIP191 data

### PR DESCRIPTION
**Pull request by magalimorin18**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -90,19 +93,15 @@ export class EIP191Signer {
     };
   }
 
-  recover(
-    message: string | Message,
-    signature: string
-  ): string {
-    const args = [].slice.apply([message, signature]);
+  recover(message: string | Message, signature: string): string {
+    // const args = [].slice.apply([message, signature]);
```
Comment (Hugoo) [inline src/index.ts]: do you want to keep this? why not just delete the line?

File: test/index.test.ts [hunk 2]
```diff
@@ -21,149 +23,187 @@ const testCases = [
...
+      );
 
-    it('should return an object with the expected properties', () => {
-      expect(signedObject).toHaveProperty('message');
-      expect(signedObject).toHaveProperty('messageHash');
-      expect(signedObject).toHaveProperty('v');
-      expect(signedObject).toHaveProperty('r');
-      expect(signedObject).toHaveProperty('s');
-      expect(signedObject).toHaveProperty('signature');
+      expect(hashedData.substring(0, 2)).toBe('0x');
```
Comment (Hugoo) [inline test/index.test.ts]: maybe a better test is to use: https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html#ishexstrict

Comment (Hugoo) [reply test/index.test.ts]: so you basically check for a hex rather than just testing for 0x prefix

File: test/index.test.ts [hunk 3]
```diff
@@ -21,149 +23,187 @@ const testCases = [
...
+  it("should throw Error with message 'Validator needs to be a valid address' when validator's address is invalid", () => {
     testCases.forEach((data) => {
-      const messageData = eip191Signer.signEthereumSignedMessage(data, signingKey);
-      const message = messageData.message;
-      const signature = messageData.signature;
-
-      const messagePrefixed = eip191Signer.signEthereumSignedMessage(message);
-      const isMessagePrefixed = true;
+      const invalidValidatorAddress =
+        '0xC1912fEE45d61C87Cc5EA59DaE31190FFFFf232d';
```
Comment (Hugoo) [inline test/index.test.ts:74]: what makes this address invalid? Wrong checksum?

Maybe using smtgh like:

```
0xImANonValidAddress
```

would make it even more obvious?

File: test/index.test.ts [hunk 4]
```diff
@@ -21,149 +23,187 @@ const testCases = [
...
+  });
+});
 
-      const recoveredAddress = eip191Signer.recover(
-        messagePrefixed,
-        signature,
-        isMessagePrefixed,
+describe('function signEthereumSignedMessage', () => {
+  it('should return an object with the expected properties', () => {
+    testCases.forEach((data) => {
```
Comment (Hugoo) [inline test/index.test.ts:92]: i'm not sure if it is necessary to perform this test against each test cases. Same below.

I mean, it is here so let's keep it, but testing this on input string `a` or `b` won't have any impact on the structure of the output. This should not change any code coverage i think.

File: test/index.test.ts [hunk 5]
```diff
@@ -21,149 +23,187 @@ const testCases = [
...
+      const messageData = eip191Signer.signEthereumSignedMessage(
+        data,
+        signingKey,
       );
+      const hasedMessage = messageData.messageHash;
+      const signature = messageData.signature;
+      const recoveredAddress = eip191Signer.recover(hasedMessage, signature);
 
       expect(recoveredAddress).toBe(signingAddress);
+      expect(recoveredAddress.substring(0, 2)).toBe('0x');
```
Comment (Hugoo) [inline test/index.test.ts]: cf above, you can use `isHexStrict`

File: test/index.test.ts [hunk 6]
```diff
@@ -21,149 +23,187 @@ const testCases = [
...
-        rawData,
-        signature,
-        false,
-      );
-      const recoveredAddressFromHashedData = eip191Signer.recover(
-        hashedData,
-        signature,
-        true,
+      expect(recoveredAddress).toBe(signingAddress);
+      expect(recoveredAddress.substring(0, 2)).toBe('0x');
```
Comment (Hugoo) [inline test/index.test.ts]: cf above, i recommend `isHexStrict`

File: README.md [hunk 7]
```diff
@@ -1,11 +1,11 @@
-# lsp6-signer.js &middot; [![GitHub license](https://img.shields.io/badge/license-Apache-blue.svg)](./LICENSE) [![npm version](https://img.shields.io/npm/v/@lukso/lsp6-signer.js.svg?style=flat)](https://www.npmjs.com/package/@lukso/lsp6-signer.js) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/lukso-network/tools-lsp6-signer/pulls)
+# eip191-signer.js &middot; [![GitHub license](https://img.shields.io/badge/license-Apache-blue.svg)](./LICENSE) [![npm version](https://img.shields.io/npm/v/@lukso/lsp6-signer.js.svg?style=flat)](https://www.npmjs.com/package/@lukso/lsp6-signer.js) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/lukso-network/tools-lsp6-signer/pulls)
 
 <p align="center">
- <h2 align="center"><strong>@lukso/lsp6-signer.js</strong></h2>
- <p align="center">Helper library to sign a transaction
+ <h2 align="center"><strong>@lukso/eip191-signer.js</strong></h2>
+ <p align="center">Helper library to sign any EIP191 data.
```
Comment (Hugoo) [inline README.md]: maybe you can add a link to: https://eips.ethereum.org/EIPS/eip-191 ? in the word EIP191.

File: README.md [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
 </p>
 
-<p align="center">For more information see <a href="https://docs.lukso.tech/tools/lsp6-signerjs/getting-started">Documentation</a>.</p>
+<p align="center">For more information see <a href="https://docs.lukso.tech/tools/eip191-signerjs/getting-started">Documentation</a>.</p>
```
Comment (Hugoo) [inline README.md:8]: make sure to also update the docs repo

File: docs/getting-started.md [hunk 9]
```diff
@@ -4,25 +4,39 @@ sidebar_position: 1
...
-The ExcuteRelayCall prefix is used instead of the standard Ethereum transaction prefix to sign messages so that an executeRelayCall transaction cannot be inadvertently signed when signing an Ethereum signed message.
+The following format is used to sign data :
 
-- [GitHub Repository](https://github.com/lukso-network/tools-lsp6-signer)
-- [NPM Package](https://www.npmjs.com/package/@lukso/lsp6-signer.js)
+```bash
+0x19 <1 byte version> <version specific data> <data to sign>
+```
+
+In the case of an Ethereum Signed Message :
```
Comment (Hugoo) [inline docs/getting-started.md]:
```suggestion
In the case of an Ethereum Signed Message:
```

File: docs/getting-started.md [hunk 10]
```diff
@@ -4,25 +4,39 @@ sidebar_position: 1
...
+
+- 1 byte version = `0x45`
+- version specific data = `thereum Signed Message:\n + len(message)`
+
+In the case of data intended to validator :
```
Comment (Hugoo) [inline docs/getting-started.md]:
```suggestion
In the case of data intended to validator:
```

intended to validator? Idk it sounds weird

**Review by Hugoo — COMMENTED** [review]
very cool

File: docs/getting-started.md (hunk 10 above)
Comment (YamenMerhi) [reply docs/getting-started.md]: In the case of data with intended validator:

File: test/index.test.ts (hunk 3 above)
Comment (frozeman) [reply test/index.test.ts:74]: I disagree, checking for a wrong checksum, but valid hex is a better test :)

File: README.md [hunk 11]
```diff
@@ -21,15 +21,193 @@ The `LSP6 ExcuteRelayCall` prefix is used instead of the standard Ethereum trans
 ## Install
 
 ```bash
-npm install @lukso/lsp6-signer.js
+npm install @lukso/eip191-signer.js
 ```
 
 ## Setup
```
Comment (frozeman) [inline README.md]: Better `Usage`
And just make  short examples in the usage section like:

```
import ()...

new class...

signEtheuem.....

recover()...
// result

signValidator...

recover()
// result
```

Lets not repeat the docs here at length. :)

**Review by frozeman — CHANGES_REQUESTED** [review]

**Comment by frozeman** [comment]
Looks good, some changes requested. Can you please also test these signatures against other tools (ethers.js? meta mask, etc) or websites that can generate them and see if we match? Same with recovering the signatures.

**Pull request merged**